### DISCUSSION TITLE
Optional JavaScript engine/DOM - Blazor project integration

### DISCUSSION THREAD

**Discussion by LukeTOBrien** [dotnet/maui#3254] [Ideas]
Hello there.

I have been working on a Blazor project for over a year, my project uses JavaScript for DOM manipulation and other things.  
I was wondering if it would would be possible for MAUI to have some kind of DOM/JavaScript engine as a NuGet package that would be able to interpret standard JavaScript APIs?  
This idea might seem a little crazy, but I am just thinking that I will have to remove my JavaScript to make it compatible with MAUL.  
As I understand, a Blazor project should be able to migrate to MAUI, but not the JavaScript.

Thanks,

Luke

**Comment by CommonLoon102** [comment]
In case of MAUI, Blazor will run inside a WebView component, which is basically a web browser without user controls. I don't see why JavaScript wouldn't work there, Blazor depends on a huge .js file itself as well.
